qemu/tests/block_during_io.py

```diff
@@ -0,0 +1,144 @@
+import re
+import logging
+import time
+
+from virttest import utils_test
+from virttest import utils_misc
+from virttest import utils_disk
+from virttest import error_context
+from provider.storage_benchmark import generate_instance
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test block devices during io.
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _get_data_disks_linux():
+        """ Get the data disks by serial or wwn options in linux. """
+        for data_image in params['images'].split()[1:]:
+            extra_params = params.get("blk_extra_params_%s" % data_image, '')
+            match = re.search(r"(serial|wwn)=(\w+)", extra_params, re.M)
+            if match:
+                drive_id = match.group(2)
+            else:
+                continue
+            drive_path = utils_misc.get_linux_drive_path(session, drive_id)
+            if not drive_path:
+                test.error("Failed to get '%s' drive path" % data_image)
+            yield drive_path[5:], params.object_params(data_image)['image_size']
+
+    def _get_data_disks_win():
+        """ Get the data disks in windows. """
+        for data_image in params['images'].split()[1:]:
+            size = params.object_params(data_image)['image_size']
+            yield utils_disk.get_windows_disks_index(session, size)[0], size
+
+    def get_data_disks():
+        """ Get the data disks. """
+        _get_disks = _get_data_disks_win if windows else _get_data_disks_linux
+        for disk, size in _get_disks():
+            yield disk, size
+
+    def configure_data_disks():
+        """ Configure the data disks. """
+        if windows:
+            utils_disk.update_windows_disk_attributes(
+                session, (disk for disk, _ in get_data_disks()))
+        return [utils_disk.configure_empty_disk(
```

Comment:
why not returning a generator expression here?

Reply:
Thanks , will update it.